### ISSUE TITLE
Issue with Enter key opening first datepicker

### ISSUE THREAD

**Issue by lukos** [WhatSock/apex#59]
Whether I have 1 or more date pickers on a page, if I press the enter key in any input box, including those that are not date pickers, it always opens the first date picker. If I press enter in the last date picker input field, it also opens the first date picker. On your examples, it doesn't do that so I'm not sure how I've broken it.

I use the following to attach the date picker to each date box input:

```javascript
$(function () {
    $('input.datebox').each(function (index, elem) {
        $A.setDatepicker({
            toggle: $(this).siblings(".date-calendar")[0],
            input: this,
            inputDateFormat: $(this).attr('placeholder')
        });
    });
});
```

This does work correctly in terms of wiring the buttons up, which do open the correct date picker and entering a date on the date picker control does put the result into the correct text box, it is just the enter key.

I found the rootListener in `4X.js` line 6699 and it seems to intercept the enter key with the `opendatepicker` event, which presumably just finds the first date picker and opens it. I can see the event being triggered explicitly in `Datepicker.js` but I don't really understand what each of these is for.

Is this something I haven't configured correctly because it seems weird?

Thanks

**Comment by accdc** [comment]
I'm having trouble diagnosing this remotely. Can you do me a favor and also include the HTML markup for the inputs and buttons that you are attaching the datepicker to?

Once I have that I'll run the code on it and see what's going on with the event attachment.

**Comment by accdc** [comment]
[Basic2.zip](https://github.com/user-attachments/files/15745678/Basic2.zip)

I created a quick example that shows how to loop through multiple datepickers on the same page, which is attached.

I recently updated the datepicker module, so downloading a new copy of the repo may be good for running this example.

Simply extract the attached zip in the folder `Templates/Datepickers`

Then use the WebserverRun.sh script in the root to start the webserver and then open the new test page at:

```
http://localhost/Templates/Datepickers/Basic2/index.htm
```

I wasn't able to reproduce the bug you mentioned, but if you notice the same thing on this example too please let me know and I'll look into it further.

**Comment by lukos** [comment]
Thanks. It is still happening but since I am currently running on a firewalled dev environment, I will need to get something into a public website so I can show you what is happening.

**Comment by lukos** [comment]
Hi @accdc sorry for the delay. If you click https://surveys-core.smartsurvey.co.uk/s/FHY5CH you will get a test survey using a new rendering engine and your date picker. This demonstrates the issue. The calendar buttons work fine to open the correct date picker instance but pressing enter in any of the text boxes always displays the first date picker.

The javascript is basically this:

```javascript
$(function () {
    $A.query('input.datebox', function (i, elem) {
        $A.setDatepicker({
            toggle: $(elem).siblings(".date-calendar")[0],
            input: elem,
            inputDateFormat: $(elem).attr('placeholder')
        });
    });
});
```

Thanks

**Comment by lukos** [comment]
You'll laugh at this. With some `debugger` statements added, I realised that it was actually the first button triggering, hence the first calendar always coming up.

It was because the calendar buttons I added did not have the type set and buttons default to "submit" which meant the enter key was nothing to do with the text inputs but was doing a form submit via the first calendar button, which then displayed the calendar and suppressed the default form submit action.

So emergency over. Thanks for your help.
